libraries/SITL/SIM_StateSpace.cpp

```diff
@@ -0,0 +1,547 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+  State Space simulator class
+*/
+
+#include "SIM_StateSpace.h"
+#include <AP_Filesystem/AP_Filesystem.h>
+
+#include <stdio.h>
+#include <sys/stat.h>
+#include <GCS_MAVLink/GCS.h>
+#include <AP_Logger/AP_Logger.h>
+
+namespace SITL {
+
+StateSpace::StateSpace(const char *frame_str) :
+    Aircraft(frame_str)
+{
+
+    
+    if (strstr(frame_str, "-heli")) {
+        frame_type = HELI_FRAME;
+
+    } else if (strstr(frame_str, "-multi")) {
+        frame_type = MULTI_FRAME;
+
+    } else {
+        frame_type = MULTI_FRAME;
+    }
+
+    const char *colon = strchr(frame_str, ':');
+    size_t slen = strlen(frame_str);
+    if (colon != nullptr && slen > 5 && strcmp(&frame_str[slen-5], ".json") == 0) {
+        load_frame_params(colon+1);
+    }
```

Comment:
I saw this function in Plane too; the part that finds a colon and a string ending in `.json`. Do you think there is a way to refactor this and reuse it?

Something like `bool is_json_frame(const char *fram_str)`